libraries/AP_ADSB/AP_ADSB.cpp

```diff
@@ -109,6 +109,28 @@ const AP_Param::GroupInfo AP_ADSB::var_info[] = {
     // @User: Advanced
     AP_GROUPINFO("RF_SELECT",   9, AP_ADSB, out_state.cfg.rfSelect, UAVIONIX_ADSB_OUT_RF_SELECT_RX_ENABLED),
 
+    // @Param: SQUAWK
+    // @DisplayName: Squawk code
+    // @Description: VFR squawk (Mode 3/A) code is a pre-programmed default code when the pilot is flying VFR and not in contact with ATC. In the USA, the VFR squawk code is 1200 [0x04B0] and in most parts of Eurpope the VFR squawk code is 7000.
+    // @Unit: decimal
+    // @User: Advanced
+    AP_GROUPINFO("SQUAWK",  10, AP_ADSB, out_state.cfg.squawk, 1200),
+
+    // @Param: FLTPLAN_ID
+    // @DisplayName: Flightplan ID
+    // @Description: This is your 8 digit flightPlanID assigned from a filed flight plan. Set to 0 if you don't have a flightplan ID.
+    // @Range: 0 99999999
+    // @User: Advanced
+    AP_GROUPINFO("FLTPLAN_ID",  11, AP_ADSB, out_state.cfg.flightplan_id, 0),
+
+    // @Param: RF_CAPABLE
+    // @DisplayName: RF capabilities
+    // @Description: Describes your hardware RF In/Out capabilities.
+    // @Values: 0:Unknown,1:Rx UAT only,3:Rx UAT and 1090ES,7:Rx&Tx UAT and 1090ES
+    // @Bitmask: 0:UAT_in,1:1090ES_in,2:UAT_out,3:1090ES_out
+    // @User: Advanced
+    AP_GROUPINFO("RF_CAPABLE",  12, AP_ADSB, out_state.cfg.rf_capable, 0),
```

Comment:
Why does the autopilot have to tell the hardware what it's capable of? This seems backwards/like a good way to get into an incorrect reporting setup.

Reply:
Yeah, I think that part is bad. I wish there was a way to ask the hardware, or not yeah it what it already knows, but this is what the uAvionix app is doing right now. They hijack the MAVLink packets to configure it. So, as of now to use the new uAvionix hardware you have to first use their app to set it up, and in some cases use it on every flight, where it would be nice to be able to do it all via ArduPilot.